racket/src/thread/schedule.rkt

```diff
@@ -28,6 +28,13 @@
 
 (define (select-thread!)
   (let loop ([g root-thread-group] [none-k maybe-done])
+    ;; check if we should collect garbage
```

Comment:
Please put these lines in a separate function that's called from here, since that will make `select-thread!` easier to read.